influxql/iterator.gen.go.tmpl

```diff
@@ -298,18 +317,11 @@ func new{{$k.Name}}SortedMergeIterator(inputs []{{$k.Name}}Iterator, opt Iterato
 		opt:    opt,
 	}
 
-	// Initialize heap.
+	// Initialize heap items.
 	for _, input := range inputs {
-		// Read next point.
-		p := input.Next()
-		if p == nil {
-			continue
-		}
-
 		// Append to the heap.
-		itr.heap = append(itr.heap, &{{$k.name}}SortedMergeHeapItem{point: p, itr: input, ascending: opt.Ascending})
+		itr.heap = append(itr.heap, &{{$k.name}}SortedMergeHeapItem{itr: input, ascending: opt.Ascending})
 	}
-	heap.Init(&itr.heap)
```

Comment:
The same reason that `MergeIterator` was changed. I moved the heap initialization to the first call of `Next()`. Part of the reason for this was so that, if an error happened, it could be returned by `Next()`.

This change isn't needed as much as the one to `MergeIterator` because the wait time for a point is less, but it should still be done that way for consistency.